src/stanc/stanc.ml

```diff
@@ -82,6 +83,9 @@ let options =
       , Arg.Set generate_inits
       , " For debugging purposes: generate a mock initial value for each \
          parameter" )
+    ; ( "--debug-data-file"
+      , Arg.String (fun s -> data_file := Some s)
+      , " For --debug-generate-data or --debug-generate-inits" )
```

Comment:
This doesn't seem to do the "right thing" for `--debug-generate-data` if you give it a partially completed JSON.

Ideally we'd make it so it does, or we'd remove the mention of it from this help string

Easy test:
```stan
data {
  int<lower=0> N;
  int<lower=0> K;
}
```
```json
{ "N":10 }
```

I'd expect the output json to be something like 
```json
{ "N": 10, "K": <any value> }
```
but I got one with `"N" : 4`

Obviously if further values depended on N it would be even more interesting if we got it right here

Reply:
Whoops, that was silly. Fixed.
I guess this is why it would be good to have tests.

Reply:
Yeah. With cram tests we could use something like `jq`:

```stan
data {
  int<lower=0> N;
  array[N] real y;
}
```
```json
{ "N": 14 }
```

and the test:
```
 stanc --debug-generate-data debug.stan --debug-data-file partial_data.json | jq -r ".y | length"
14
```